### DISCUSSION TITLE
Is there a way to get the screen's/connection's name

### DISCUSSION THREAD

**Discussion by nathannino** [qtile/qtile#4899] [Q&A] [answered]
I've been slowly writing a qtile config, and for reasons that are not important to this post, I've decided to use eww instead of qtile's built-in widget system. Today, I have realized that the `--screen` argument in eww, when given a number, is putting it on the wrong screen. After testing, I can put it on the screen I expect it to be in using the connector instead, so `--screen HDMI-0` for example.

But looking at the qtile docs, I don't see any way to get that name inside of qtile itself. Am I mistaken? If I am, how do I get the screen/connection inside of qtile? If I am not, is there something I can use instead to get that name sent to qtile?

**Comment by nathannino** [comment]
So, while the question stands for getting the screen/connection name inside of qtile, I did figure out a workaround to link it between the order given by `xrandr --listmonitors` and the order inside qtile. I don't like it, as it is basically both a regex of a command output and the link between the screen/connection name and the qtile screen object is it's x and y coordinates, so it doesn't feel like it should be stable, but it seems to work for now?

I don't really want to post it, because I am *not* that good at python yet and the code itself is messy, but I can share it if asked.

Just want to reiterate that this does not close my question, as I would still rather have a way to do this without relying on screen coordinates and external commands, if possible

**Reply by ervinpopescu** [reply, marked as answer]
Qtile's `Screen` object does not have the info you ask for, yet.

See https://docs.qtile.org/en/latest/manual/config/screens.html, #4693, https://github.com/tych0/qtile/tree/serial-number-ordering.

[This commit](https://github.com/tych0/qtile/commit/2074b9a196dd336fffc8a9cec48cddc55c1f8f12) is probably what you are looking for.

**Reply by nathannino** [reply]
I had a look at the commit sent, and I learned that the connection name is not stable. Fortunately, my current configuration only requires it to be stable for as long as the monitor is plugged in, but having a serial that stays stable between reboots would give me more options in the future.

For now, I will keep my screen position hack, but I will definitely be waiting until this or something similar is merged and released so that I can migrate over to the cleaner and more stable option.
